### PR TITLE
Modernize job strings into DEFINES

### PR DESCRIPTION
**Pull request by EvilDragonfiend**
## About The Pull Request
Modernize job strings into DEFINES

Things you should know
1.  Lower case job strings should be exceptional from this unless we put something like `lower(JOB_ASSISTANT)`
2. HTML or TGUI related strings appear using DEFINES is not legit.
3. I am not very sure, but TG or Skyrat didn't use defines at some points (i.e. the 2 one above, but more than it). I used my perception to guess which should be kept in raw strings.

## Why It's Good For The Game
refactor good

## Testing Photographs and Procedure
This is really difficult to test in a single person's environment. this needs testmerge in actual rounds.
although, in my environment, It was fine anyways, but you know, it's difficult to check everything at a single person's side....

## Changelog
:cl:
refactor: refactored job strings into DEFINES.
refactor: relocated some job related code lines
refactor: changed some abbreviation names into correct ones (i.e. hos -> head_of_security)
/:cl:

### CODE REVIEW

File: code/__DEFINES/jobs.dm [hunk 1]
```diff
@@ -99,11 +99,107 @@
...
+#define JOB_CYBORG "Cyborg"
+#define JOB_PERSONAL_AI "Personal AI"
+//Security
+#define JOB_WARDEN "Warden"
+#define JOB_DETECTIVE "Detective"
+#define JOB_SECURITY_OFFICER "Security Officer"
+#define JOB_SECURITY_OFFICER_MEDICAL "Security Officer (Medical)"
+#define JOB_SECURITY_OFFICER_ENGINEERING "Security Officer (Engineering)"
+#define JOB_SECURITY_OFFICER_SCIENCE "Security Officer (Science)"
+#define JOB_SECURITY_OFFICER_SUPPLY "Security Officer (Cargo)"
```
Comment (EvilDragonfiend) [inline code/__DEFINES/jobs.dm]: these are needed to port ID and access subsystem in future.

File: code/game/machinery/PDApainter.dm [hunk 2]
```diff
@@ -10,71 +10,71 @@
...
+		JOB_CARGO_TECHNICIAN = "pda-cargo",
+		JOB_CHAPLAIN = "pda-chaplain",
+		JOB_CHEMIST = "pda-chemistry",
+		JOB_CHIEF_MEDICAL_OFFICER = "pda-cmo",
+		JOB_CHIEF_ENGINEER = "pda-ce",
+		JOB_CLOWN = "pda-clown",
+		JOB_COOK = "pda-cook",
+		JOB_CURATOR = "pda-library",
+		JOB_DETECTIVE = "pda-detective",
 		"Engineer" = "pda-engineer",
```
Comment (Penwin0) [inline code/game/machinery/PDApainter.dm]: Missed one? if not on purpose you also missed Scientist here

Comment (EvilDragonfiend) [reply code/game/machinery/PDApainter.dm]: This `Engineer` means both of Station engi and atmos tech (like the Service one). but scientist is a good catch. thanks.

**Comment by EvilDragonfiend** [comment]
Due to hud icon PR, this is temporarily closed as draft to prevent merge conflict.

File: code/__HELPERS/jobs.dm [hunk 3]
```diff
@@ -83,75 +83,75 @@
 	var/static/id_to_hud = list(
 		// Command
 		"Command (Custom)" = JOB_HUD_RAWCOMMAND,
-		"Captain" = JOB_HUD_CAPTAIN,
+		JOB_NAME_HEADOFPERSONNEL = JOB_HUD_CAPTAIN,
```
Comment (PowerfulBacon) [inline code/__HELPERS/jobs.dm]: This is wrong

**Review by PowerfulBacon — CHANGES_REQUESTED** [review]

File: code/__HELPERS/jobs.dm (hunk 3 above)
Comment (EvilDragonfiend) [reply code/__HELPERS/jobs.dm]: oops typo

**Review by PowerfulBacon — APPROVED** [review]
Seems okay

**Pull request merged**